src/main/java/de/dennisguse/opentracks/services/TrackDeleteService.java

```diff
@@ -137,35 +122,28 @@ public TrackDeleteService getService() {
     }
 
     public static class DeleteStatus {
-        private final int progress;
-        private final int max;
-        private final Track.Id trackId;
+        private final List<Track.Id> trackIds;
 
         /**
-         * @param trackId  Track.Id just deleted if any.
-         * @param progress number of deletes.
-         * @param max      total of deletes to be done.
+         * @param trackIds List of deleted Track.Ids.
          */
-        private DeleteStatus(@Nullable Track.Id trackId, int progress, int max) {
-            this.trackId = trackId;
-            this.progress = progress;
-            this.max = max;
+        private DeleteStatus(@Nullable List<Track.Id> trackIds) {
+            this.trackIds = trackIds;
         }
 
         public boolean isFinished() {
-            return progress == max;
+            return true;
         }
 
         public boolean isDeleted(Track.Id trackId) {
-            return this.trackId != null && this.trackId.equals(trackId);
+            return this.trackIds != null && this.trackIds.contains(trackId);
```

Comment:
Can also be removed, right?

Reply:
The `TrackRecordedActivity` is checking it. It could be the case that the currently recorded track was in the list of to be deleted tracks, but was filtered out, because it is not allowed to delete it.